packages/event-handler/tests/unit/appsync-graphql/AppSyncGraphQLResolver.test.ts

```diff
@@ -0,0 +1,174 @@
+import context from '@aws-lambda-powertools/testing-utils/context';
+import {
+  onMutationEventFactory,
+  onQueryEventFactory,
+} from 'tests/helpers/factories.js';
+import { beforeEach, describe, expect, it, vi } from 'vitest';
+import { AppSyncGraphQLResolver } from '../../../src/appsync-graphql/AppSyncGraphQLResolver.js';
+import { ResolverNotFoundException } from '../../../src/appsync-graphql/errors.js';
+
+describe('Class: AppSyncGraphQLResolver', () => {
```

Comment:
We should add a test where we don't provide a logger, just to ensure that our fallback works:

```ts
it('logs only warnings and errors using global console object if no logger supplied', async () => {
const app = new AppSyncGraphQLResolver();

// rest of test
})

```

Reply:
Also we should have a test to verify we respect the environment variable `AWS_LAMBDA_LOG_LEVEL` being set to `DEBUG`. Example here: https://github.com/aws-powertools/powertools-lambda-typescript/blob/d124369db6ec63c9610ea4a01c30bee5560f4d8d/packages/event-handler/tests/unit/bedrock-agent/BedrockAgentFunctionResolver.test.ts#L98